toonz/sources/toonzqt/spreadsheetviewer.cpp

```diff
@@ -21,60 +22,94 @@
+             &FrameScroller::onVScroll);
+  disconnect(m_scrollArea->horizontalScrollBar(), &QScrollBar::valueChanged,
+             this, &FrameScroller::onHScroll);
+}
+
+void FrameScroller::connectScrollbars() {
+  if (!m_scrollArea) return;
+  m_lastX = m_scrollArea->horizontalScrollBar()->value();
+  m_lastY = m_scrollArea->verticalScrollBar()->value();
+  connect(m_scrollArea->verticalScrollBar(), &QScrollBar::valueChanged, this,
+          &FrameScroller::onVScroll);
+  connect(m_scrollArea->horizontalScrollBar(), &QScrollBar::valueChanged, this,
+          &FrameScroller::onHScroll);
 }
 
-bool FrameScroller::isScrollerConnected(FrameScroller *scroller) {
-  return m_connectedScrollers.contains(scroller);
+void FrameScroller::onVScroll(int y) {
+  QPoint offset(0, y - m_lastY);
+  m_lastY = y;
+  handleScroll(offset);
+}
+void FrameScroller::onHScroll(int x) {
+  QPoint offset(0, x - m_lastX);
+  m_lastX = x;
+  handleScroll(offset);
 }
 
 static QList<FrameScroller *> frameScrollers;
 
+void FrameScroller::handleScroll(const QPoint &offset) const {
+  CellPositionRatio ratio = orientation()->xyToPositionRatio(offset);
+  if (!ratio.frame())  // only synchronize changes by frames axis
+    return;
+
+  for (int i = 0; i < frameScrollers.size(); i++)
+    if (frameScrollers[i] != this) frameScrollers[i]->onScroll(ratio);
+}
+
+void adjustScrollbarWithoutSignals(QScrollBar *scrollBar, int add);
+
+void FrameScroller::onScroll(const CellPositionRatio &ratio) {
+  QPoint offset = orientation()->positionRatioToXY(ratio);
+  if (offset.x())
+    adjustScrollbarWithoutSignals(m_scrollArea->horizontalScrollBar(),
+                                  offset.x());
+  if (offset.y())
+    adjustScrollbarWithoutSignals(m_scrollArea->verticalScrollBar(),
+                                  offset.y());
+}
+void adjustScrollbarWithoutSignals(QScrollBar *scrollBar, int add) {
+  scrollBar->blockSignals(true);
+  scrollBar->setValue(scrollBar->value() + add);
+  scrollBar->blockSignals(false);
+}
+
 void FrameScroller::registerFrameScroller() {
-  if (!frameScrollers.contains(this)) {
-    for (int i = 0; i < frameScrollers.size(); i++)
-      connectScroller(frameScrollers[i]);
-    frameScrollers.append(this);
-  }
+  if (!frameScrollers.contains(this)) frameScrollers.append(this);
 }
 
 void FrameScroller::unregisterFrameScroller() {
-  if (frameScrollers.contains(this)) {
-    frameScrollers.removeAll(this);
-    for (int i = 0; i < frameScrollers.size(); i++)
-      disconnectScroller(frameScrollers[i]);
-  }
+  if (frameScrollers.contains(this)) frameScrollers.removeAll(this);
 }
 
-void FrameScroller::prepareToScroll(int dy) {
-  if (dy == 0) return;
+void FrameScroller::prepareToScrollOthers(const QPoint &offset) {
+  CellPositionRatio ratio = orientation()->xyToPositionRatio(offset);
   for (int i = 0; i < frameScrollers.size(); i++)
-    if (frameScrollers[i] != this) frameScrollers[i]->onPrepareToScroll(dy);
```

Comment:
Fixed. Since the timeline columns are narrower, coordinates have to be adjusted accordingly on other scrollbars depending on which scroll bar triggered the event.  Had to correct logic to get the other panes to sync properly without causing more scrolling events.